tests/JsonSchema.test.ts

```diff
@@ -0,0 +1,518 @@
+      expect(
+        base.isJsonIntersection({
+          allOf: [{ type: 'string' }, { type: 'number' }],
+        }),
+      ).toBe(true)
+    })
+  })
+
+  describe('derivation', () => {
+    const testSchema = S.Readonly(
+      S.Struct({
+        literal: S.Literal('string', 5, true, null),
+        nully: S.Nullable(S.Int({ min: 0, max: 1 })),
+        partial: S.Partial({
+          optNull: S.OptionFromNullable(S.Float({ min: 0, max: 1 })),
+          optUndefined: S.OptionFromUndefined(S.Float({ min: 0, max: 1 })),
+        }),
+        rec: S.Record(S.CreditCard),
+        arr: S.Array(S.Json.jsonString),
+
+        tup: S.Tuple(S.JsonFromString, S.Number),
+        sum: S.Sum('type')({
+          a: S.Struct({ type: S.Literal('a'), a: S.Boolean }),
+          b: S.Struct({ type: S.Literal('b'), b: S.Lazy('Sum[b].b', () => S.Natural) }),
+        }),
+        intersection: S.Intersection(S.Struct({ a: S.NonPositiveInt }))(
+          S.Struct({ b: S.NonNegativeFloat, c: S.NonPositiveFloat }),
+        ),
+        date: S.Date.dateFromString,
+        isoDate: S.DateFromIsoString(),
+        map: S.Map(Str.Ord, S.String, S.Float()),
+        int: S.Int(),
+        option: S.Option(69420, S.Number),
+        singularSum: S.Sum('type')({
+          a: S.Struct({
+            type: S.Literal('a'),
+            a: S.NonEmptyArray(
+              S.Padding.padRight(
+                { by: 'ExactLength', exactLength: 1 },
+                '*',
+              )(
+                S.Padding.padRight(
+                  { by: 'MaxLength', maxLength: 2 },
+                  '=',
+                )(
+                  S.Padding.padLeft(
+                    { by: 'MaxLength', maxLength: 3 },
+                    '=',
+                  )(
+                    S.Padding.padLeft(
+                      { by: 'ExactLength', exactLength: 4 },
+                      '=',
+                    )(
+                      S.Padding.padRight(
+                        { by: 'ExactLength', exactLength: () => 0 },
+                        '&',
+                      )(
+                        S.Padding.padRight(
+                          { by: 'MaxLength', maxLength: () => 0 },
+                          '=',
+                        )(
+                          S.Padding.padLeft(
+                            { by: 'MaxLength', maxLength: () => 0 },
+                            '=',
+                          )(
+                            S.Padding.padLeft(
+                              { by: 'ExactLength', exactLength: () => 0 },
+                              '=',
+                            )(S.String),
+                          ),
+                        ),
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          }),
+        }),
+      }),
+    )
+    const jsonSchema = getJsonSchema(testSchema)
+    const testValue = JSON.parse(JSON.stringify(jsonSchema)) as any
```

Comment:
:thinking: why the JSON parse/stringify? wouldn't `jsonSchema` already be the expected jsonSchema? I thought maybe it was to do a test `expect(jsonSchema).toEqual(testValue)` but I don't see that one anywhere

Reply:
It's related to using classes under the hood.  I think the classes are nice for if they're console.log'd because instead of `Object{ }` it'll say `JsonString{ }` for example.  But Jest also looks for instance equality too.  So this basically strips the class identity